src/OrchardCore.Modules/OrchardCore.Setup/Startup.cs

```diff
@@ -1,20 +1,59 @@
 using System;
+using System.Linq;
 using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 using OrchardCore.Modules;
 
 namespace OrchardCore.Setup
 {
     public class Startup : StartupBase
     {
+        public Startup(IHostingEnvironment env)
+        {
+            var builder = new ConfigurationBuilder()
```

Comment:
@jtkech do we need this code to be able to read the configuration from the main appsettings.json? I don't think so.

@hishamco it should work by resolving `IShellConfiguration` directly. Look for an example in the Media module.

Reply:
Normally not when resolving `IShellConfiguration` but better to try it in the context of the setup module.

Then, you don't need to resolve the `OrchardCore` section.